pkg/apis/serving/v1alpha2/framework_pytorch.go

```diff
@@ -47,6 +47,7 @@ func (s *PyTorchSpec) GetContainer(modelName string, config *InferenceServicesCo
 			"--model_name=" + modelName,
 			"--model_class_name=" + s.ModelClassName,
 			"--model_dir=" + constants.DefaultModelLocalMountPath,
+			"--http_port=" + constants.GetInferenceServiceHttpPort(hasLogging),
```

Comment:
It's a little bit crufty to wire through the logging component into all of our concepts. What do you think about using a more "sidecar"-y approach.

We could have a mutator that injects a second container into the workload and remaps the ports using: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#containerport-v1-core.

This way, from the perspective of the predictor, it's still serving on the same port. From the perspective of the pod sepc, we can wire up our logger to sit where knative thinks it should be (i.e. 8080) and call out to the user containers new port (i.e. 8081?). All of these details can be fully encapsulated inside the sidecar injector.

WDYT?

Reply:
I don't think this is possible as all containers share a network on the pod so ports can't be conflicting. `Number of port to expose on the pod's IP address` - note reference to Pod here.

Reply:
Knative default to use 8080 but can specify one port by the containerport in spec.
https://github.com/knative/serving/blob/24a4448a9d520e02aef1ebc3d041826d2b3819ff/pkg/reconciler/revision/resources/deploy.go#L174-L181

So I think we need not do it in injector or remap it. We can just make the logger always listen on 8081 and the predictor always on 8080, but check the hasLogger condition and if true add this when we reconcile knative service
```
          ports:
            - name: http1
              containerPort: 8081
```
In this way although we still need the haslogger condition but we need not redefine the port, and also we need not do it in every predictor/transformer/explainer, but can just add this on the top level of knative service resource.

Reply:
This seems like a good idea to clean the integration. As we are not covering Custom containers.

Reply:
I have implemented this idea in PR #592  , please take a look.